streams/service/src/main/java/com/hortonworks/streamline/streams/service/SchemaResource.java

```diff
@@ -102,55 +109,27 @@ public Response postStreamsSchema(StreamsSchemaInfo streamsSchemaInfo,
         return WSUtils.respondEntity(schemaIdVersion, OK);
     }
 
-    // This API would change once we consider other sources. Currently supports kafka sources for the given topic names.
-    @GET
-    @Path("/{topicName}")
-    @Timed
-    @Produces(MediaType.APPLICATION_JSON)
-    public Response getKafkaSourceSchema(@PathParam("topicName") String topicName,
-                                         @Context SecurityContext securityContext) throws JsonProcessingException {
-        try {
-            LOG.info("Received path: [{}]", topicName);
-            // for now, takes care of kafka for topic values. We will enhance to work this to get schema for different
-            // sources based on given properties.
-            final String schemaName = topicName;
-            SchemaVersionInfo schemaVersionInfo = schemaRegistryClient.getLatestSchemaVersionInfo(schemaName);
-            String schema = schemaVersionInfo != null ? schemaVersionInfo.getSchemaText() : null;
-            LOG.debug("######### Received schema from schema registry: ", schema);
-            if (schema != null && !schema.isEmpty()) {
-                schema = AvroStreamlineSchemaConverter.convertAvroSchemaToStreamlineSchema(schema);
-            }
-            LOG.debug("######### Converted schema: {}", schema);
-            return WSUtils.respondEntity(schema, OK);
-        } catch (SchemaNotFoundException e) {
-            LOG.error("Schema not found for topic: [{}]", topicName, e);
-            throw EntityNotFoundException.byId(topicName);
-        } catch (JsonProcessingException ex) {
-            LOG.error("Error occurred while retrieving schema with name [{}]", topicName, ex);
-            throw ex;
-        } catch (Exception e) {
-            throw new RuntimeException(e);
-        }
-    }
-
     @GET
     @Path("/{schemaName}/versions")
     @Timed
     @Produces(MediaType.APPLICATION_JSON)
     public Response getAllSchemaVersions(@PathParam("schemaName") String schemaName,
                                          @Context SecurityContext securityContext) {
+        return doGetAllSchemaVersionForBranch(schemaName, null);
+    }
+
+    private Response doGetAllSchemaVersionForBranch(String schemaName, String branchName) {
         try {
             LOG.info("Get all versions for schema : {}", schemaName);
-            Collection<SchemaVersionInfo> schemaVersionInfos = schemaRegistryClient.getAllVersions(schemaName);
+            List<Byte> states = Collections.singletonList(SchemaVersionLifecycleStates.ENABLED.getId());
```

Comment:
@satishd One other question i had after refreshing on SR branching feature is that we allow a schema branch version in a "in review" or other custom state to be used by producer/consumer. Here, we are just returning ENABLED schemas. Is there a reason for that or should we be allowing schema versions in those states?

Reply:
@priyank5485 Only enabled schema versions are allowed, any other state is not considered ready to be consumed when a topology is configured. Users use non master branches to try out the changes by enabling that version and merge it to master only if they find that version is working as expected in their topology.

Reply:
Based on this document https://docs.google.com/document/d/1qi3o9ADxkySZn-PZJiWTyfcvdBraW-dHGwMRm-xZeyA/edit# it seems that Archived, ChangeRequired, etc states are where it is available to producers/consumers. Given that SAM Kafka components are producers/consumers, should not we allow them to play with a schema on a branch which is under review?

Also, based on the document it seems that once a schema is enabled it can be merged to master. If its anyway mergeable to master then why expose the branches of a schema in SAM? We can go back to old UI and on server side we just get versions for the master branch. I guess the main question is "Are we allowing SAM users to try out schemas that are on a branch and under review?" If not and if we only want enabled schemas(that are anyway mergeable to master) then this PR should have only changes for server side.

Reply:
@priyank5485 

>Also, based on the document it seems that once a schema is enabled it can be merged to master. If its anyway mergeable to master then why expose the branches of a schema in SAM? 

It depends on how are verification workflows are designed in the applications. There are different ways users want to use lifecycle states. That is the reason SR client APIs give flexible APIs to fetch schema versions with whatever states they are interested in. Default or custom review states can be considered outside of SAM and it is enabled only when it can be tried in applications like SAM. If that application fails with this schema then it can be disabled and eventually discarded. Enabled schema does not need to be always merged. Merge operation can be done based on the result of trying out on SAM or any other respective application.

So, it depends on how users want to try out branched schema versions. I am OK to allowing non-master branch schema versions with states other than archived/deleted state in SAM.